### PR TITLE
Fix #1375 - Highlighting currently selected page on sidebar broken with sections

### LINKED ISSUE #1375: Highlighting currently selected page on sidebar broken with sections
**Current behavior**

Highlighting currently selected section works well with [basic](https://github.com/styleguidist/react-styleguidist/tree/master/examples/basic) example:

image

However when using [sections](https://github.com/styleguidist/react-styleguidist/tree/master/examples/sections) example it doesn't work anymore:

image

`Components` in sidebar is not highlighted.

**To reproduce**

Run `sections` example: https://github.com/styleguidist/react-styleguidist/tree/master/examples/sections

**Expected behavior**

`Components` is highlighted.

### PR DESCRIPTION
**Pull request by jkarttunen**
Fix with IF condition. Check if routing is based on flat or sectioned system.

### CODE REVIEW

File: src/client/rsg-components/ComponentsList/ComponentsListRenderer.js [hunk 1]
```diff
@@ -37,14 +37,18 @@ const styles = ({ color, fontFamily, fontSize, space, mq }) => ({
...
+	const { pagePerSection } = config;
 	items = items.filter(item => item.visibleName);
 
 	if (!items.length) {
 		return null;
 	}
 
-	const windowHash = window.location.pathname + getHash(window.location.hash);
+	// Match selected page for both routing conventions
+	const hash = window.location.hash;
```
Comment (sapegin) [inline src/client/rsg-components/ComponentsList/ComponentsListRenderer.js]:
```suggestion
	const {hash, pathname} = window.location;
```

File: src/client/rsg-components/ComponentsList/ComponentsListRenderer.js [hunk 2]
```diff
@@ -37,14 +37,18 @@ const styles = ({ color, fontFamily, fontSize, space, mq }) => ({
...
+export function ComponentsListRenderer({ classes, items }, { config }) {
+	const { pagePerSection } = config;
 	items = items.filter(item => item.visibleName);
 
 	if (!items.length) {
 		return null;
 	}
 
-	const windowHash = window.location.pathname + getHash(window.location.hash);
+	// Match selected page for both routing conventions
```
Comment (sapegin) [inline src/client/rsg-components/ComponentsList/ComponentsListRenderer.js]: Please make this comment more specific: which both?

File: src/client/rsg-components/ComponentsList/ComponentsListRenderer.js (hunk 1 above)
Comment (jkarttunen) [reply src/client/rsg-components/ComponentsList/ComponentsListRenderer.js]: Ok!

File: src/client/rsg-components/ComponentsList/ComponentsListRenderer.js (hunk 2 above)
Comment (jkarttunen) [reply src/client/rsg-components/ComponentsList/ComponentsListRenderer.js]: Best i could come up with was
// Match selected component in both basic routing and pagePerSection routing.

Comment (sapegin) [reply src/client/rsg-components/ComponentsList/ComponentsListRenderer.js]: Already much better 🦄

**Comment by sapegin** [comment]
Merging, thanks! 🌮

**Pull request merged**